runtime/Java/src/org/antlr/v4/runtime/misc/IntegerList.java

```diff
@@ -57,6 +58,8 @@ public IntegerList(int capacity) {
 		if (capacity < 0) {
 			throw new IllegalArgumentException();
 		}
+        if (capacity > MAX_ARRAY_SIZE)
+          throw new IllegalArgumentException();
```

Comment:
:bulb: Considering `ArrayList` omits this check, I think it's safe for us to as well.